test cases/common/41 test args/meson.build

```diff
@@ -33,3 +33,23 @@ testfilect = custom_target('testfile',
                            build_by_default : true,
                            command : [copy, '@INPUT@', '@OUTPUT@'])
 test('custom target arg', tester, args : testfilect, env : env_array)
+
+# Same tests as above, but with positional args instead of kwarg
+test('command line arguments - positional', e1, 'first', 'second')
+test('custom target arg - positional', tester, testfilect, env : env_array)
+
+# Command can be a python script
+test('python script', 'script.py')
+
+# ExternalProgram can be passed as positional args
+py = find_program('python3')
+test('program in args', 'script.py', py)
```

Comment:
This reads in a confusing way. It would probably be better to require that the first argument be a target or found program. It also seems that the script does not actually do anything with its argument, it should at least verify that it gets the expected input (i.e. `sys.argv[1]`).

Reply:
I changed a bit that test, hopes it's clearer now. That test checks 2 new behaviours:
- A test executable can be passed as string, it does find_program() on first arg by default like for most other functions. Note that `test('mytest', find_program('script.py'))` was working previously, so that's just a shorthand for something already possible.
- Found external programs can be passed in test args. Granted that's probably not really useful, but the point here is I'm trying to make a common list of allowed types in commands across all our APIs for consistency.